Updated to 1.19 + fixed/added? durability

ChristianKilduff opened the pull request:
My friend wanted the umbrellas to have durability so I updated it to 1.19 and added durability for them. I could not figure out what you were trying to do in the tick injection. I just cleared out the original code and wrote it so that the umbrella takes damage if its held in sunlight or in rain. I also removed the entity mixin as it appeared to me to do nothing anymore, I presume it was an old way of dealing with rain.

ChristianKilduff committed:
Updated to 1.19 + durability

updated mod to 1.19 as well as added/fixed? durability to umbrella.

ChristianKilduff committed:
updated version + archive base name

ChristianKilduff committed:
added comment to origins_version

ChristianKilduff committed:
removed unused methods

ChristianKilduff committed:
Made read me not default\

ChristianKilduff committed:
forgot to update title

ChristianKilduff committed:
updated mc version

ChristianKilduff committed:
added netherite_umbrellas, just unbreakable umbrellas
changed damage speed and switched .damage to .setdamage

ChristianKilduff committed:
added slight netherite umbrella textures

ChristianKilduff committed:
made umbrellas breakable

Fusion-Flux commented:
umbreallas are not supposed to break they are ment to no longer work when they reach no durability

ChristianKilduff commented:
> umbreallas are not supposed to break they are ment to no longer work when they reach no durability

yea the other pull request seemed to understand the goal of durability better than me, in my repo, I removed the checks for durability from other parts of the mod and also my code does not have the wet repair. I just prefer it with standard durability.

When the durability just acts as a wetness meter, the only origins that get debuffed r the ones who cannot touch rain. But the umbrella serves a purpose for both sun and rain. Which basically puts sun avoiding individuals at a distinct advantage with umbrellas which is not smth I want. I figured if thats what u want, the other pull had it covered and I could just use my version with my friend. Altho I would say think about the balance. I think the wetness meter is rlly cool concept, but in practice just makes the umbrella worse for certain origins while easier for others even tho they both use it. Even the sun avoiders would use the umbrella more as there is more sun than rain. + I added netherite umbrella cause I like the idea of an endgame umbrella when there is standard durability and made a temporary texture and my friend was gonna design it to look prettier at some point. which might not be the direction u want to go either.

ChristianKilduff committed:
Update PlayerEntityMixin.java

make umbrellas last longer

ChristianKilduff committed:
fixed tags not working and also made umbrellas have a tag so that both umbrellas function properly
